sql/src/main/java/org/apache/apex/malhar/sql/codegen/BeanClassGenerator.java

```diff
@@ -0,0 +1,771 @@
+    setterNodeSec.instructions
+      .add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, "java/lang/Integer", "intValue", "()I", false));
+    setterNodeSec.instructions.add(new InsnNode(Opcodes.I2L));
+    setterNodeSec.instructions.add(new LdcInsnNode(new Long(1000)));
+    setterNodeSec.instructions.add(new InsnNode(Opcodes.LMUL));
+    setterNodeSec.instructions
+      .add(new MethodInsnNode(Opcodes.INVOKESPECIAL, "java/lang/Long", "<init>", "(J)V", false));
+    setterNodeSec.instructions
+      .add(new FieldInsnNode(Opcodes.PUTFIELD, classNode.name, fieldName + "Ms", "Ljava/lang/Long;"));
+
+    setterNodeSec.instructions.add(new InsnNode(Opcodes.RETURN));
+    classNode.methods.add(setterNodeSec);
+
+    // Create setter for Ms
+    MethodNode setterNodeMs = new MethodNode(Opcodes.ACC_PUBLIC, "set" + fieldNameForMethods + "Ms",
+        "(Ljava/lang/Long;)V", null, null);
+    setterNodeMs.instructions.add(new VarInsnNode(Opcodes.ALOAD, 0));
+    setterNodeMs.instructions.add(new TypeInsnNode(Opcodes.NEW, type));
+    setterNodeMs.instructions.add(new InsnNode(Opcodes.DUP));
+    setterNodeMs.instructions.add(new VarInsnNode(Opcodes.ALOAD, 1));
+    setterNodeMs.instructions
+      .add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, "java/lang/Long", "longValue", "()J", false));
+    setterNodeMs.instructions.add(new MethodInsnNode(Opcodes.INVOKESPECIAL, "java/util/Date", "<init>", "(J)V", false));
+    setterNodeMs.instructions.add(new FieldInsnNode(Opcodes.PUTFIELD, classNode.name, fieldName, "L" + type + ";"));
+
+    setterNodeMs.instructions.add(new VarInsnNode(Opcodes.ALOAD, 0));
+    setterNodeMs.instructions.add(new TypeInsnNode(Opcodes.NEW, "java/lang/Integer"));
+    setterNodeMs.instructions.add(new InsnNode(Opcodes.DUP));
+    setterNodeMs.instructions.add(new VarInsnNode(Opcodes.ALOAD, 1));
+    setterNodeMs.instructions
+      .add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, "java/lang/Long", "longValue", "()J", false));
+    setterNodeMs.instructions.add(new LdcInsnNode(new Long(1000)));
+    setterNodeMs.instructions.add(new InsnNode(Opcodes.LDIV));
+    setterNodeMs.instructions.add(new InsnNode(Opcodes.L2I));
+    setterNodeMs.instructions
+      .add(new MethodInsnNode(Opcodes.INVOKESPECIAL, "java/lang/Integer", "<init>", "(I)V", false));
+    setterNodeMs.instructions
+      .add(new FieldInsnNode(Opcodes.PUTFIELD, classNode.name, fieldName + "Sec", "Ljava/lang/Integer;"));
+
+    setterNodeMs.instructions.add(new VarInsnNode(Opcodes.ALOAD, 0));
+    setterNodeMs.instructions.add(new VarInsnNode(Opcodes.ALOAD, 1));
+    setterNodeMs.instructions
+      .add(new FieldInsnNode(Opcodes.PUTFIELD, classNode.name, fieldName + "Ms", "Ljava/lang/Long;"));
+
+    setterNodeMs.instructions.add(new InsnNode(Opcodes.RETURN));
+    classNode.methods.add(setterNodeMs);
+  }
+
+  @SuppressWarnings("unchecked")
+  private static void addIntGetterNSetter(ClassNode classNode, String fieldName, String fieldNameForMethods,
```

Comment:
Except return opcodes I don't see any other differences. And the this method here is used for all byte, boolean, char and int type which is already confusing. I would prefer Yogi's suggestion